libbeat/scripts/kibana_extract.py

```diff
@@ -0,0 +1,144 @@
+
+import os
+import json
+import jinja2
+
+
+def unescape_json(escaped):
+    obj = json.loads(escaped)
+    return json.dumps(obj, sort_keys=True, indent=4, separators=(',', ': '))
+
+def load_json(filepath):
+    json_data=open(filepath).read()
+    return json.loads(json_data)
+
+def load_search_source(file):
+    data = load_json(file)
+
+    field = "searchSourceJSON"
+
+    if data.has_key("kibanaSavedObjectMeta") and data["kibanaSavedObjectMeta"].has_key(field):
+        jsonData = unescape_json(data["kibanaSavedObjectMeta"][field])
+    else:
+        return ""
+
+    write_output(file, field, jsonData)
+
+
+def load_field(file, field):
+    data = load_json(file)
+
+    if  data.has_key(field):
+        jsonData = unescape_json(data[field])
+    else:
+        return ""
+
+    write_output(file, field, jsonData)
+
+
+def write_output(file, field, jsonData):
+
+    dir_name = os.path.splitext(file)[0]
+
+    if not os.path.exists(dir_name):
+        os.mkdir(dir_name)
+
+    with open(dir_name + "/" + field + ".json", 'w') as outfile:
+        outfile.write(jsonData)
+
+
+def extract():
+    base = './etc/kibana'
+
+    folders = os.listdir(base)
```

Comment:
Another option would be to first go over all dashboards of that Beat (starting with the name of the Beat by convention). Then parse the json file to get the list of visualizations by extracting the `savedSearchId` value. Then search for the visualizations needed in the bunch of visualizations (from all Beats), parse the visualizations to extract the list of searches that are used and then look for those searches.